--- kinetic-c-0.6.0/Makefile.orig	2014-10-07 00:54:46.000000000 +0200
+++ kinetic-c-0.6.0/Makefile	2014-10-14 22:04:07.222209100 +0200
@@ -12,9 +12,10 @@
 SOCKET99 = ./vendor/socket99
 VND_INC = ./vendor
 BIN = $(BIN_DIR)/kinetic_client
-LDFLAGS += -lm -l ssl -l kinetic-c-client -l crypto
+LDFLAGS += -lm -lssl -lcrypto -lsocket99 -lprotobuf-c
 
 PREFIX ?= /usr/local
+LIB ?= lib
 INSTALL ?= install
 RM ?= rm
 
@@ -39,10 +40,9 @@
 LIB_INCS = -I$(LIB_DIR) -I$(PUB_INC) -I$(PBC_INC) -I$(VND_INC)
 LIB_DEPS = $(PUB_INC)/kinetic_client.h $(PUB_INC)/kinetic_types.h $(LIB_DIR)/kinetic_connection.h $(LIB_DIR)/kinetic_hmac.h $(LIB_DIR)/kinetic_logger.h $(LIB_DIR)/kinetic_message.h $(LIB_DIR)/kinetic_nbo.h $(LIB_DIR)/kinetic_operation.h $(LIB_DIR)/kinetic_pdu.h $(LIB_DIR)/kinetic_proto.h $(LIB_DIR)/kinetic_socket.h
 # LIB_OBJ = $(patsubst %,$(OUT_DIR)/%,$(LIB_OBJS))
-LIB_OBJS = $(OUT_DIR)/kinetic_nbo.o $(OUT_DIR)/kinetic_operation.o $(OUT_DIR)/kinetic_pdu.o $(OUT_DIR)/kinetic_proto.o $(OUT_DIR)/kinetic_socket.o $(OUT_DIR)/kinetic_message.o $(OUT_DIR)/kinetic_logger.o $(OUT_DIR)/kinetic_hmac.o $(OUT_DIR)/kinetic_connection.o $(OUT_DIR)/kinetic_types.o $(OUT_DIR)/kinetic_client.o $(OUT_DIR)/socket99.o $(OUT_DIR)/protobuf-c.o
+LIB_OBJS = $(OUT_DIR)/kinetic_nbo.o $(OUT_DIR)/kinetic_operation.o $(OUT_DIR)/kinetic_pdu.o $(OUT_DIR)/kinetic_proto.o $(OUT_DIR)/kinetic_socket.o $(OUT_DIR)/kinetic_message.o $(OUT_DIR)/kinetic_logger.o $(OUT_DIR)/kinetic_hmac.o $(OUT_DIR)/kinetic_connection.o $(OUT_DIR)/kinetic_types.o $(OUT_DIR)/kinetic_client.o
 
-# default: $(KINETIC_SO)
-default: $(KINETIC_LIB)
+default: $(KINETIC_LIB) $(KINETIC_SO)
 
 test_all: Rakefile $(LIB_OBJS)
 	@echo
@@ -101,7 +101,7 @@
 	@echo --------------------------------------------------------------------------------
 	@echo Building $(KINETIC_SO) dynamic library
 	@echo --------------------------------------------------------------------------------
-	$(CC) $(LIB_OBJS) -shared ${LDFLAGS} -o ${KINETIC_SO}
+	$(CC) $(LIB_OBJS) -shared ${LDFLAGS} -Wl,-soname,${KINETIC_SO} -o ${KINETIC_SO}
 
 libso: $(KINETIC_SO)
 
@@ -170,7 +171,7 @@
 
 # Installation
 # Need to re-add ${KINETIC_SO}, once ready
-install: ${KINETIC_LIB} VERSION
+install: ${KINETIC_SO} ${KINETIC_LIB} VERSION
 	@echo
 	@echo --------------------------------------------------------------------------------
 	@echo Installing $(PROJECT) v$(VERSION) into $(PREFIX)
@@ -178,15 +179,13 @@
 	@echo
 	@echo You may be prompted for your password in order to proceed.
 	@echo
-	${INSTALL} -d ${PREFIX}/lib/
-	${INSTALL} -c ${KINETIC_LIB} ${PREFIX}/lib/
-	# ${INSTALL} -c ${KINETIC_SO} ${PREFIX}/lib/
+	${INSTALL} -d ${PREFIX}/${LIB}/
+	${INSTALL} -c ${KINETIC_LIB} ${PREFIX}/${LIB}/
+	${INSTALL} -c ${KINETIC_SO} ${PREFIX}/${LIB}/ ; ln -s ${KINETIC_SO} ${PREFIX}/${LIB}/libkinetic-c-client.so
 	${INSTALL} -d ${PREFIX}/include/
 	${INSTALL} -c ./include/${API_NAME}.h ${PREFIX}/include/
 	${INSTALL} -c ./include/kinetic_types.h ${PREFIX}/include/
 	${INSTALL} -c ./src/lib/kinetic_proto.h ${PREFIX}/include/
-	${INSTALL} -d ${PREFIX}/include/protobuf-c
-	${INSTALL} -c ./vendor/protobuf-c/protobuf-c/protobuf-c.h ${PREFIX}/include/protobuf-c/
 
 uninstall:
 	@echo
--- ../BUILD/kinetic-c-0.6.0/Makefile.orig	2014-10-17 18:57:11.958468832 +0200
+++ ../BUILD/kinetic-c-0.6.0/Makefile	2014-10-17 21:53:09.594692437 +0200
